src/lib/y2network/sysconfig/hostname_writer.rb

```diff
@@ -73,8 +73,15 @@ def update_sysconfig_dhcp(hostname, old_hostname)
       #
       # @param hostname [Y2Network::Hostname] Hostname configuration
       def update_hostname(hostname)
-        Yast::Execute.on_target!("/usr/bin/hostname", hostname.hostname.split(".")[0])
-        Yast::SCR.Write(Yast::Path.new(".target.string"), HOSTNAME_PATH, "#{hostname.hostname}\n")
+        hostname = hostname.static
+        # 1) when user asked for ereasing hostname from /etc/hostname, we keep runtime as it is
```

Comment:
```suggestion
        # 1) when user asked for erasing hostname from /etc/hostname, we keep runtime as it is
```

Reply:
In case of erasing and no dhcp, shouldn't it be `localhost`?

Reply:
Note: We probably should use hostnamectl in most of the cases, and systemd-firstboot during installation, but it is something we can discuss about.

Reply:
well, empty hostname is treated as invalid by hostname utility and empty /etc/hostname is automatically changed to ```localhost``` during the boot.

However, ```hostname ''``` even ```hostname -b --file /etc/hostname``` do not work (expected in first case, but not so expected in the second one according to the man pages). So, runtime remains untouched (hostname cmd still returns previous hostname).

```hostnamectl set-hostname ''``` works in a sense that it deletes /etc/hostname, erases static hostname and sets transient hostname to localhost, so ```hostname``` cmd then correctly shows ```localhost``` as result. However, things like bash promt remains untouched (which is also true when modifying hostname to a nonempty string).

We can put localhost directly into /etc/hostname to mimic the behavior but in my POV it is something different than user wanted to achieve.

So, we can do exactly what hostnamectl does - delete ```/etc/hostname```. However, until we switch to ```hostnamectl ``` we most probably should warn user by a popup that runtime hostname stays untouched until reboot..